modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java

```diff
@@ -37,6 +38,18 @@
     public DartClientCodegen() {
         super();
 
+        additionalReservedWords.addAll(
+                Sets.newHashSet(
```

Comment:
Thinking about the opposite problem. I don't generate ApiClient class, I would generate only models. This makes ApiClient a non-reserved word, thus produces unexpected results.

The problem exists, but I'm not convinced this is the right solution here.

Reply:
I think this is the correct solution, doing this as well in the dio generator. You may have a couple false positives but that is ok I'd say.

Reply:
There is the option to only generate the models? Then just add the classes to the set only if the parameter is not set. Do you mind telling me the option name?

Reply:
>Or if you have another idea just tell us, I'm open to better solutions

Well, no. It would have to be conditionally added increasing complexity of such solution. I would prefer to see global PR for this problem rather than each generator doing it on their own.

@wing328 is there any DefaultCodeGen solution for this? A set of _potentially_ used keywords like `ApiClient` to prevent generated files collide with project files. It would have to be managed independently by each code generator and their cmd options.

Looks like big effort to handle some small cases, but correctness of the output should be priority(?).

Reply:
> is there any DefaultCodeGen solution for this? A set of _potentially_ used keywords like `ApiClient` to prevent generated files collide with project files. It would have to be managed independently by each code generator and their cmd options.

The problem is that most other generators/languages have a concept of absolute/fq imports but Dart does not.
So in other generators you can use the import mapping which can be configured as cmd parameters or in the generator class. The dart generators basically miss uses the import mapping, meaning we HAVE TO use it differently due to the absence of the absolute/fq imports concept.

The `additionalReservedWords ` is just something I came up with for the dio generator to solve this in a simple manner.

Reply:
Well that is also not fully correct. Dart has named imports `import "package:foo:foo.dart" as bar;` but I tried this and it fails with a lot of builders/build_runner (`json_serializable`, `built_value`).